abipy/lumi/deltaSCF.py

```diff
@@ -0,0 +1,606 @@
+
+        return d
+    
+    def get_dataframe_species(self):
+        """
+        Panda dataframe with relevant properties per species.
+        units used are 
+        mass     - amu
+        DeltaR   - Angstrom
+        DeltaQ^2 - amu.Angstrom^2
+        DeltaF   - eV/Angstrom
+        """
+        list_of_dict=[]
+        for index,specie in enumerate(self.structuregs.types_of_species):
+            d=self.get_dict_per_specie(specie)
+            list_of_dict.append(d)
+
+        return pd.DataFrame(list_of_dict)
+
+
+    def delta_r(self):
+        """
+        Total Delta_R (Angstrom)
+        """
+        d_r_squared=np.sum(self.diff_pos()**2)
+        return(np.sqrt(d_r_squared))
+
+    def amu_list(self):
+        """
+        List of masses (amu)
+        """
+        amu_list=[]
+        for atom in self.structuregs.species:
+            amu_list.append(atom.atomic_mass)
+        return(amu_list)
+
+    def delta_q(self,unit='atomic'):
+        """
+        Total Delta_Q 
+        Args:
+            unit: amu^1/2.Angstrom if unit = 'atomic', kg^1/2.m if 'SI'
+        """
+        sq_Q_matrix = np.zeros((self.natom(), 3))
+        for a in np.arange(self.natom()):
+            for i in np.arange(3):
+                sq_Q_matrix[a, i] = self.amu_list()[a] * self.diff_pos()[a, i] ** 2
+        delta_Q = np.sqrt(np.sum(sq_Q_matrix))
+
+        if unit == 'SI':
+            return (delta_Q* 1e-10 * np.sqrt(1.66053892173E-27))
+        else:
+            return(delta_Q)
+
+    def effective_mass(self):
+        """
+        Effective mass M
+        """
+        M = self.delta_q() ** 2 / self.delta_r() ** 2
+        return (M)
+
+    def E_zpl(self):
+        """
+        Zero-phonon line energy (eV)
+        """
+        return (self.AestarEnergy - self.AgEnergy)
+
+    def E_em(self):
+        """
+        Emission energy(eV)
+        """
+        return(self.AestarEnergy-self.AeEnergy)
+
+    def E_abs(self):
+        """
+        Absorption energy(eV)
+        """
+        return(self.AgstarEnergy-self.AgEnergy)
+
+    def E_FC_ex(self,unit='eV'):
+        """
+        Franck condon energy in excited state (eV)
+        = Relaxation energy between Ag* and Ae* states
+        """
+        e_fc=self.AgstarEnergy - self.AestarEnergy
+        if unit == 'SI':
+            return 1.602176565e-19*e_fc
+        else:
+            return e_fc
+
+    def E_FC_gs(self,unit='eV'):
+        """
+        Franck condon energy in ground state (eV)
+        = Relaxation energy between Ae and Ag states
+        """
+        e_fc=self.AeEnergy - self.AgEnergy
+        if unit == 'SI':
+            return 1.602176565e-19*e_fc
+        else:
+            return e_fc
+
+    def Stoke_shift(self):
+        """
+        Stokes shift (eV)
+        """
+        return(self.E_FC_ex()+self.E_FC_gs())
+
+    def eff_freq_gs(self):
+        """
+        Phonon effective frequency of the ground state (eV)
+        """
+        hbar_eV = 6.582119570e-16  # in eV*s
```

Comment:
hbar_eV can be moved to abipy.core.abinit_units and then used in deltaSCF.py so that we avoid multiple definitions.